src/main/java/net/fabricmc/loader/impl/game/minecraft/Log4jLogHandler.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2016 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.loader.impl.game.minecraft;
+
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import net.fabricmc.loader.util.log.Log;
+import net.fabricmc.loader.util.log.LogCategory;
+import net.fabricmc.loader.util.log.LogHandler;
+import net.fabricmc.loader.util.log.LogLevel;
+
+final class Log4jLogHandler implements LogHandler {
+	@Override
+	public boolean shouldLog(LogLevel level, LogCategory category) {
+		return getLogger(category).isEnabled(translateLogLevel(level));
+	}
+
+	@Override
+	public void log(long time, LogLevel level, LogCategory category, String msg, Throwable exc) {
+		getLogger(category).log(translateLogLevel(level), msg, exc);
+	}
+
+	private static Logger getLogger(LogCategory category) {
+		Logger ret = (Logger) category.data;
```

Comment:
Why is this kept as a data field on category object than a map in this implementation?

Reply:
Saves lookup time and keeps it simpler, the field is specifically for this kind of use.